src/extension_commands.ml

```diff
@@ -534,6 +534,143 @@ module Copy_type_under_cursor = struct
+                   ~detail:(Option.value ~default:"" res.doc)
+                   ()
+               , res.name )))
+        ~options:
+          (QuickPickOptions.create
+             ~title:"Type/Polarity Search Results"
+             ~canPickMany:false
+             ~ignoreFocusOut:true
+             ())
+        ())
+
+  let _search_by_type =
+    let handler (instance : Extension_instance.t) ~args:_ =
+      let search_by_type () =
+        match Window.activeTextEditor () with
+        | None ->
+          Extension_consts.Command_errors.text_editor_must_be_active
+            extension_name
+            ~expl:"The command to search for a value by it's type"
+          |> show_message `Error "%s" |> Promise.return
+        | Some text_editor -> (
+          match Extension_instance.lsp_client instance with
+          | None ->
+            show_message `Warn "ocamllsp is not running" |> Promise.return
+          | Some (_client, ocaml_lsp)
+            when not (Ocaml_lsp.can_handle_search_by_type ocaml_lsp) ->
+            ocaml_lsp_doesnt_support_search_by_type instance ocaml_lsp
+            |> Promise.return
+          | Some (client, _) -> (
+            let open Promise.Syntax in
+            let position =
+              TextEditor.selection text_editor |> Selection.active
+            in
+            let* query_input =
+              Vscode.(
+                Window.showInputBox
+                  ~options:
+                    (InputBoxOptions.create
+                       ~title:"Search By Type"
+                       ~prompt:
+                         "Perform a search by type request by providing a type \
+                          signature to look for"
+                       ~placeHolder:"int -> string"
+                       ~value:"int -> string"
+                       ~password:false
+                       ~ignoreFocusOut:true
+                       ())
+                  ())
+            in
+            match query_input with
+            | Some query -> (
+              let* type_search_results =
+                get_search_results
+                  ~query
+                  ~with_doc:true
+                  ~limit:100
+                  ~position
+                  text_editor
+                  client
+              in
+              let* type_picker = display_search_results type_search_results in
+              match type_picker with
+              | Some text ->
+                let+ type_inserted =
+                  TextEditor.edit
+                    text_editor
+                    ~callback:(fun ~editBuilder ->
+                      TextEditorEdit.insert
+                        editBuilder
+                        ~location:position
+                        ~value:text)
+                    ()
+                in
+                if not type_inserted then
+                  show_message `Error "Unable to insert %s" text
+                else show_message `Info "Inserted %s" text
```

Comment:
For this command, messages should only appear to signal errors, the user will see in the editor that the insertion worked.